framework/src/org/checkerframework/common/fieldtypeinference/FieldTypeInferenceTransfer.java

```diff
@@ -0,0 +1,112 @@
+package org.checkerframework.common.fieldtypeinference;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Modifier;
+
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.FieldAccess;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.RegularTransferResult;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.UnderlyingAST;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.dataflow.cfg.node.LocalVariableNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * 
+ * This Transfer Function performs type inference for private fields in a class.
+ * <p>
+ * To enable type inference for private fields, the Checker's ATF must be a subtype of
+ * <tt>org.checkerframework.common.fieldtypeinference.FieldTypeInferenceAnnotatedTypeFactory</tt>.
+ *
+ * @author pbsf
+ */
+
+public class FieldTypeInferenceTransfer extends CFTransfer {
+
+    private final Map<Receiver, AnnotatedTypeMirror> assignedPrivateFields;
```

Comment:
Mike, I removed this map as suggested by Jonathan.

Jonathan, thanks for the good practice tip. I moved addAssignedField to the AnnotatedTypeFactory, and also created in the ATF the methods getAssignedPrivateFieldsKeySet and getAssignedPrivateFieldATM to make the map immutable (except by calling the addAssignedField method).